test/e2e/utils/chaise.page.js

```diff
@@ -873,6 +873,25 @@ function chaisePage() {
     this.waitForElementCondition = function(condition, timeout) {
         return browser.wait(condition, timeout || browser.params.defaultTimeout);
     };
+
+    this.performLogin = function(cookie, defer) {
+
+        defer = defer || require('q').defer();
+
+        browser.get(process.env.CHAISE_BASE_URL + "/login/");
+        browser.ignoreSynchronization = true;
+          
+        browser.wait(protractor.ExpectedConditions.visibilityOf(element(by.id("loginApp"))), browser.params.defaultTimeout).then(function() {
+            browser.driver.executeScript('document.cookie="' + cookie + ';path=/;' + (process.env.TRAVIS ? '"' : 'secure;"')).then(function() {
+              browser.ignoreSynchronization = false;
+              defer.resolve();
+            });
+        }, function() {
+            defer.reject();
```

Comment:
Not necessarily. As this would mean that the test is unable to open the login page due to some server issue. Due to which all the tests will fail. 

We can add, but uptil now I have never encountered that issue.